### PR TITLE
Filter functionality

### PR DESCRIPTION
**Pull request by anuj-kumary**
Following are the filters added:
- Rating
- Price Range
- Categories
- Sort By Price

ScreenShot:-
Screenshot from 2022-03-17 13-07-13
Screenshot from 2022-03-17 13-07-41

### CODE REVIEW

File: eagle-stores/src/App.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import "./App.css";
-import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import { Routes, Route } from 'react-router-dom';
 import NavBar from "./components/NavBar/NavBar"
```
Comment (znjs) [inline eagle-stores/src/App.js]: Suggestion: can use barrel imports

File: eagle-stores/src/App.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-import { Home } from "./pages";
-import Carousel from "./components/Carousel/Carousel"
-import HomeCategory from "./components/HomeCategory/HomeCategory";
+import { Home, Product } from "./pages";
 import Footer from "./components/Footer/Footer";
```
Comment (znjs) [inline eagle-stores/src/App.js]: Suggestion: can use barrel imports

File: eagle-stores/src/context/data/data-context.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
+import { createContext, useContext, useEffect , useReducer } from "react";
+import {ACTION_TYPE} from "../../utils/actionType"
+import { initialistate , DataReducer } from "../../reducer/DataReducer";
```
Comment (znjs) [inline eagle-stores/src/context/data/data-context.js]: Suggestion: can define a `index.js` and export from it directly

File: eagle-stores/src/index.js [hunk 4]
```diff
@@ -1,14 +1,22 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import App from "./App";
+import { BrowserRouter } from 'react-router-dom';
 import { makeServer } from "./server";
+import { DataProvider } from "./context/data/data-context";
```
Comment (znjs) [inline eagle-stores/src/index.js:6]: Suggestion: can use barrel export and import directly from `./context`

File: eagle-stores/src/pages/Product/Product.jsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { ProductAside } from "./components/ProductAside/ProductAside";
+import { ProductMain } from "./components/ProductMain/ProductMain";
```
Comment (znjs) [inline eagle-stores/src/pages/Product/Product.jsx:2]: Suggestion: can use barrel imports

File: eagle-stores/src/pages/Product/components/ProductMain/ProductMain.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import "./ProductMain.css";
+import { useData } from "../../../../context/data/data-context";
+import { ProductCard } from "../ProductCard/ProductCard";
```
Comment (znjs) [inline eagle-stores/src/pages/Product/components/ProductMain/ProductMain.jsx:3]: Suggestion: can use barrel imports

File: eagle-stores/src/utils/filterFunction.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  if (rating === "") return data;
+  const rate = Number(rating);
+  return data.filter((item) => Number(item.rating) >= rate);
+};
+
+const sortByPrice = (data, sortBy) => {
+  if (sortBy === "") return data;
+  if (sortBy === "HighToLow") {
+    return [...data].sort((a, b) => b.price - a.price);
+  } else if (sortBy === "LowToHigh")
```
Comment (znjs) [inline eagle-stores/src/utils/filterFunction.js]: Suggestion: can do an early return and remove `else if(...)`

**Review by znjs — APPROVED** [review]
Clean code
Good naming conventions
Good file structure

**Review by samyakshah3008 — APPROVED** [review]
3 things I liked about this PR: 
Loved how you set up your folder structure 
Clean code gave better readability 
Good  naming conventions as well   

Suggestion: You can also include some comments where necessary in future

File: eagle-stores/src/App.css [hunk 8]
```diff
@@ -1,51 +1,8 @@
...
-}
-
-@media (prefers-reduced-motion: no-preference) {
-  .App-logo {
-    animation: App-logo-spin infinite 20s linear;
-  }
-}
-
-.App-header {
-  background-color: #22201c;
```
Comment (Rohanmond) [inline eagle-stores/src/App.css:17]: use variable

File: eagle-stores/src/App.js [hunk 9]
```diff
@@ -18,14 +16,13 @@ function MockAPI() {
...
-    <Home />
-    <Footer />
     <Routes>
-      <Route path="mock-man" element={<Mockman />} />
+      <Route path="/" element={ <Home />} />
+      <Route path="/product" element={<Product />} />
+      <Route path="/mockman" element={<Mockman />} />
     </Routes>
-   </BrowserRouter>
+    <Footer />
```
Comment (Rohanmond) [inline eagle-stores/src/App.js:24]: Very clean App.js

File: eagle-stores/src/components/NavBar/NavBar.jsx [hunk 10]
```diff
@@ -18,11 +19,11 @@ const NavBar = () => {
 
         <ul className="navbar__right">
         <div>
-        <a
+        <Link to="/product"
```
Comment (Rohanmond) [inline eagle-stores/src/components/NavBar/NavBar.jsx:22]: better if you use useNavigate here

File: eagle-stores/src/App.js (hunk 2 above)
Comment (anuj-kumary) [reply eagle-stores/src/App.js]: Done

File: eagle-stores/src/index.js [hunk 11]
```diff
@@ -1,14 +1,22 @@
...
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+  
+  <DataProvider>
+  <BrowserRouter>
```
Comment (Rohanmond) [inline eagle-stores/src/index.js:13]: put BrowserRouter at the top

File: eagle-stores/src/utils/filterFunction.js (hunk 7 above)
Comment (anuj-kumary) [reply eagle-stores/src/utils/filterFunction.js]: Done

File: eagle-stores/src/pages/Product/Product.jsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+export const Product = () => {
+  return (
+    <>
+      <div className="product__container">
+        <ProductAside />
+        <ProductMain />
```
Comment (Rohanmond) [inline eagle-stores/src/pages/Product/Product.jsx:9]: very clean productlist

File: eagle-stores/src/pages/Product/components/ProductMain/ProductMain.jsx (hunk 6 above)
Comment (anuj-kumary) [reply eagle-stores/src/pages/Product/components/ProductMain/ProductMain.jsx:3]: Done

File: eagle-stores/src/utils/actionType.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+export const ACTION_TYPE = {
+    INITIALIZE_PRODUCTS : "INITIALIZE_PRODUCTS",
+    INITIALIZE_CATEGORIES: "INITIALIZE_CATEGORIES",
+    filterChange: "filterChange",
```
Comment (Rohanmond) [inline eagle-stores/src/utils/actionType.js]: better to follow only one convention on constants. Don't use different Convensions

**Review by Rohanmond — APPROVED** [review]

File: eagle-stores/src/components/NavBar/NavBar.jsx (hunk 10 above)
Comment (anuj-kumary) [reply eagle-stores/src/components/NavBar/NavBar.jsx:22]: Thanks for suggestion i will work on that

File: eagle-stores/src/utils/actionType.js (hunk 13 above)
Comment (anuj-kumary) [reply eagle-stores/src/utils/actionType.js]: Done

File: eagle-stores/src/index.js (hunk 11 above)
Comment (anuj-kumary) [reply eagle-stores/src/index.js:13]: Done

**Pull request merged**